vividus-extension-selenium/src/main/java/org/vividus/ui/context/UiContext.java

```diff
@@ -39,32 +40,26 @@
     @Override
     public SearchContext getSearchContext()
     {
-        SearchContextData searchContext = testContext.get(KEY, SearchContextData.class);
-        if (searchContext != null)
-        {
-            return searchContext.getSearchContext();
-        }
-        return null;
+        return getSearchContext(SearchContext.class).orElse(null);
     }
 
     @Override
-    public <T extends SearchContext> T getSearchContext(Class<T> clazz)
+    public <T extends SearchContext> Optional<T> getSearchContext(Class<T> clazz)
     {
-        SearchContext searchContext = getSearchContext();
+        SearchContextData searchContextData = testContext.get(KEY, SearchContextData.class);
+        SearchContext searchContext = searchContextData != null ? searchContextData.getSearchContext() : null;
```

Comment:
When searchContextData could be null?

Reply:
when user tries to run UI step without starting the application

Reply:
I think we could track it, right? If the  web driver initialized